src/detours.cpp

```diff
@@ -122,31 +122,41 @@ struct _DETOUR_TRAMPOLINE
     _DETOUR_ALIGN   rAlign[8];      // instruction alignment array.
     PBYTE           pbRemain;       // first instruction after moved code. [free list]
     PBYTE           pbDetour;       // first instruction of detour function.
+#ifdef DETOURS_SUPPORT_DETOUR_CONTEXT_CALLBACK
     BYTE            rbContext[20];  // context callback code (calls context function w/ void* param)
     PBYTE           pbContext;      // first instruction of context callback code (rbSetContext) if used
+#endif // DETOURS_SUPPORT_DETOUR_CONTEXT_CALLBACK
 };
 
-C_ASSERT(sizeof(_DETOUR_TRAMPOLINE) == 96);
+#ifdef DETOURS_SUPPORT_DETOUR_CONTEXT_CALLBACK
+C_ASSERT(sizeof(_DETOUR_TRAMPOLINE) == 96);
+#else
+C_ASSERT(sizeof(_DETOUR_TRAMPOLINE) == 72);
+#endif // DETOURS_SUPPORT_DETOUR_CONTEXT_CALLBACK
 
 enum {
     SIZE_OF_JMP = 5
-};
-
-inline PBYTE detour_gen_context_callback(PBYTE pbCode, PF_DETOUR_INVOKE_CONTEXT_CALLBACK pfContextCallback, PVOID pContextParam)
-{
-    PBYTE pbJmpSrc = pbCode + 15;
-
-    *pbCode++ = 0x68;   // push param
-    *((INT32*&)pbCode)++ = (INT32)(pContextParam);
-
-    *pbCode++ = 0x68;   // push return address
-    *((INT32*&)pbCode)++ = (INT32)(pbJmpSrc);
-
-    *pbCode++ = 0xE9;   // jmp +imm32  (jmp fnSetContext)
-    *((INT32*&)pbCode)++ = (INT32)((PBYTE)pfContextCallback - pbJmpSrc);
-
-    return pbCode;
-}
+};
+
+#ifdef DETOURS_SUPPORT_DETOUR_CONTEXT_CALLBACK
+
+inline PBYTE detour_gen_context_callback(PBYTE pbCode, PF_DETOUR_INVOKE_CONTEXT_CALLBACK pfContextCallback, PVOID pContextParam)
+{
+    PBYTE pbJmpSrc = pbCode + 15;
+
+    *pbCode++ = 0x68;   // push param
+    *((INT32*&)pbCode)++ = (INT32)(pContextParam);
+
+    *pbCode++ = 0x68;   // push return address
+    *((INT32*&)pbCode)++ = (INT32)(pbJmpSrc);
+
+    *pbCode++ = 0xE9;   // jmp +imm32  (jmp fnSetContext)
+    *((INT32*&)pbCode)++ = (INT32)((PBYTE)pfContextCallback - pbJmpSrc);
+
+    return pbCode;
+}
+
+#endif // DETOURS_SUPPORT_DETOUR_CONTEXT_CALLBACK
```

Comment:
nit: I always dislike repetition FOO_SUPPORT_FOO_BAR
should be FOO_SUPPORT_BAR imho.